### PR TITLE
#2234 remove ufRaw continuous variable from laod restorative with lim…

### LINKED ISSUE #2234: Reduce number of variables for load restorative cpp model

### PR DESCRIPTION
**Pull request by marcochiaramello**
…its model

Closes #2234

### CODE REVIEW

**Comment by marcochiaramello** [comment]
@cossartque fyi

File: dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp [hunk 1]
```diff
@@ -157,80 +153,95 @@ namespace DYN {
...
 
   void
   ModelLoadRestorativeWithLimits::setFequations() {
     if (isConnected()) {
-      fEquationIndex_[0] = std::string("Tf_*UfRawPrim - (U - UfRaw ):");
-      fEquationIndex_[1] = std::string("Uf - UfRaw:");
-      fEquationIndex_[2] = std::string("Ir - (P * Ur + Q * Ui) / (U * U): ");
-      fEquationIndex_[3] = std::string("Ii - (P * Ui - Q * Ur) / (U * U): ");
+      if (!UMaxPuReached_ && !UMinPuReached_) {
+        fEquationIndex_[0] = std::string("Tf_*UfPrim - (U - Uf ):");
```
Comment (rosiereflo) [inline dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp:235]: Please remove the useless "_"

File: dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp [hunk 2]
```diff
@@ -157,80 +153,95 @@ namespace DYN {
...
+  }
   void
   ModelLoadRestorativeWithLimits::evalStaticYType() {
-    yType_[0] = DIFFERENTIAL;  // differential equation on ufRaw
-    yType_[1] = ALGEBRAIC;  // uf
-    yType_[2] = EXTERNAL;  // ur
-    yType_[3] = EXTERNAL;  // ui
-    yType_[4] = ALGEBRAIC;  // ir
-    yType_[5] = ALGEBRAIC;  // ii
+    yType_[1] = EXTERNAL;  // ur
```
Comment (rosiereflo) [inline dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp:174]: Please use the index defined in ::init here

File: dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/test/TestLoadRestorativeWithLimits.cpp [hunk 3]
```diff
@@ -287,17 +291,27 @@ TEST(ModelsLoadRestorativeWithLimits, ModelLoadRestorativeWithLimitsContinuousAn
   ASSERT_NO_THROW(modelLoad->evalZ(0));
   ASSERT_NO_THROW(modelLoad->evalMode(0));
   ASSERT_NO_THROW(modelLoad->setGequations());
+  ASSERT_NO_THROW(modelLoad->evalDynamicYType());
+  ASSERT_EQ(yType[0], ALGEBRAIC);
```
Comment (rosiereflo) [inline dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/test/TestLoadRestorativeWithLimits.cpp:295]: Is there a possibility to get the DIFFERENTIAL type in the tests?

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp [hunk 4]
```diff
@@ -264,10 +275,12 @@ namespace DYN {
   void
   ModelLoadRestorativeWithLimits::evalJt(const double /*t*/, const double cj, SparseMatrix& jt, const int rowOffset) {
     if (!isConnected()) {
-      jt.changeCol();  //  ufRaw
-      jt.addTerm(UfRawYNum_ + rowOffset, cj);
-      jt.changeCol();  //  uf
-      jt.addTerm(UfYNum_ + rowOffset, 1);
+      jt.changeCol();  // uf
+      if (!UMaxPuReached_ && !UMinPuReached_) {
```
[the remark this answers is not in the document]
Comment (marcochiaramello) [reply dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp:279]: my bad, thx

File: dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp (hunk 1 above)
Comment (marcochiaramello) [reply dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp:235]: solved

File: dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp (hunk 2 above)
Comment (marcochiaramello) [reply dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp:174]: the initialization of indexes is now done at constructor, as init is called after evalStaticYType

File: dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/test/TestLoadRestorativeWithLimits.cpp (hunk 3 above)
Comment (marcochiaramello) [reply dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/test/TestLoadRestorativeWithLimits.cpp:299]: done

**Review by rosiereflo — APPROVED** [review]

File: dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp [hunk 5]
```diff
@@ -140,12 +137,7 @@ namespace DYN {
 
   void
   ModelLoadRestorativeWithLimits::init(const double /*t0*/) {
-    UfRawYNum_ = 0;
-    UfYNum_ = 1;
-    UrYNum_ = 2;
-    UiYNum_ = 3;
-    IrYNum_ = 4;
-    IiYNum_ = 5;
+    /* not need */
```
Comment (cossartque) [inline dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp:140]: not needed

File: nrt/data/DynaFlow/LoadRestorativeWithLimits/reference/outputs/curves/curves.csv [hunk 6]
```diff
@@ -1,34 +1,34 @@
-time;_Load__10_PPu_value;_Load__10_QPu_value;NETWORK__BUS___10_TN_Upu_value;_Load__10_UfRawPu_value;_Load__10_UfPu_value;
```
Comment (cossartque) [inline nrt/data/DynaFlow/LoadRestorativeWithLimits/reference/outputs/curves/curves.csv:1]: How does this change affect the nrt of dynawo-rte ?

Comment (marcochiaramello) [reply nrt/data/DynaFlow/LoadRestorativeWithLimits/reference/outputs/curves/curves.csv:1]: very small changes in values

File: dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp (hunk 5 above)
Comment (marcochiaramello) [reply dynawo/sources/Models/CPP/Components/ModelLoadRestorativeWithLimits/DYNModelLoadRestorativeWithLimits.cpp]: done, thx

**Review by cossartque — APPROVED** [review]

**Pull request merged**